tests/test_xdp_container_restart.sh

```diff
@@ -0,0 +1,222 @@
+    local image_name="$1"
+    local containerfile="$2"
+    
+    if ! "${OCI_BIN}" build -f "${containerfile}" . --tag "${image_name}"; then
+        echo "Failed to build image ${image_name} from container file ${containerfile} with ${OCI_BIN}" >&2
+        exit 1
+    fi
+    
+    local image_id
+    image_id=$(get_container_id "${image_name}")
+    if [[ -z "${image_id}" ]]; then
+        echo "Failed to get image id for ${image_name}" >&2
+        exit 1
+    fi
+    
+    echo "${image_id}"
+}
+
+# Function to run new container
+run_new_container() {
+    local container_name="$1"
+    local image_id="$2"
+    local port_mapping="$3"  # Optional, can be empty
+    local privileged="$4"    # true/false
+    local entrypoint="$5"    # Optional, can be empty
+    shift 5
+    local command=("$@")     # Remaining arguments as command array
+    
+    local args=("run" "--name" "${container_name}")
+    
+    # Add port mapping if provided
+    if [[ -n "${port_mapping}" ]]; then
+        args+=("-p" "${port_mapping}")
+    fi
+    
+    # Add privileged if true
+    if [[ "${privileged}" == "true" ]]; then
+        args+=("--privileged")
+    fi
+    
+    # Add entrypoint if provided
+    if [[ -n "${entrypoint}" ]]; then
+        if [[ "${entrypoint}" == " " ]]; then
+            entrypoint=""
+        fi
+        args+=("--entrypoint" "${entrypoint}")
+    fi
+    
+    # Add tmpfs mount
+    args+=("--mount" "type=tmpfs,dst=/run")
+    
+    # Add detached mode and image
+    args+=("-d" "${image_id}")
+    
+    # Add command if provided
+    if [[ ${#command[@]} -gt 0 ]]; then
+        args+=("${command[@]}")
+    fi
+    
+    local container_id
+    container_id=$("${OCI_BIN}" "${args[@]}")
+    
+    if [[ -z "${container_id}" ]]; then
+        echo "Failed to run container ${container_name} with ${OCI_BIN}, got no container id" >&2
+        exit 1
+    fi
+    
+    echo "${container_id}"
+}
+
+# Create the commit name from a provided name
+get_commit_name() {
+    local name="$1"
+    echo "${name}_commit"
+}
+
+# Function to execute command in container
+exec_in_container() {
+    local container_id="$1"
+    local cmd="$2"
+    
+    local output
+    if ! output=$("${OCI_BIN}" exec "${container_id}" bash -c "${cmd}" 2>&1); then
+        echo "Command failed: ${output}" >&2
+        exit 1
+    fi
+    
+    echo "${output}"
+}
+
+# Function to commit container to image
+commit_container() {
+    local container_id="$1"
+    local new_image_name="$2"
+    
+    if ! "${OCI_BIN}" commit "${container_id}" "${new_image_name}"; then
+        echo "Failed to commit container" >&2
+        exit 1
+    fi
+}
+
+# Main test function
+main() {
+    echo "Starting XDP container restart test..."
+    
+    # Start OCI service (docker/podman)
+    if [[ "${OCI_BIN}" == "docker" ]]; then
+        if ! systemctl is-active docker; then
+            echo "Starting docker service..."
+            if ! systemctl start docker; then
```

Comment:
Shouldn't we additionally wait after starting? 

(untested)

```sh
echo "Waiting for docker service to be active..."
if ! systemctl is-active --quiet --wait docker; then
    echo "Docker service failed to start" >&2
    exit 1
fi

echo "Waiting for docker daemon to be ready..."
timeout="${DOCKER_TIMEOUT:-30}"

while [ "$timeout" -gt 0 ]; do
    if docker info >/dev/null 2>&1; then
        echo "Docker daemon is ready"
        break
    fi
    sleep 1
    ((timeout--))
done

if [ "$timeout" -eq 0 ]; then
    echo "Docker daemon failed to become ready within ${DOCKER_TIMEOUT:-30} seconds" >&2
    exit 1
fi
```

Reply:
ok done and I think it's ready subject to passing the tests